src/test/unit/common/proxy.test.ts

```diff
@@ -14,22 +16,74 @@ suite('Proxy', () => {
   const protocol = 'https:';
   const proxy = `${protocol}//${auth}@${host}:${port}`;
   const proxyStrictSSL = true;
+  const error = sinon.stub();
+  const logger = { error } as unknown as ILog;
 
   teardown(() => {
     sinon.restore();
   });
 
   test('No proxy set', () => {
     const getConfiguration = sinon.stub();
+    const getInsecure = sinon.stub();
+    getInsecure.returns(!proxyStrictSSL);
+
+    const configuration = {
+      getInsecure,
+    } as unknown as IConfiguration;
+
     getConfiguration.withArgs('http', 'proxy').returns(undefined);
 
     const workspace = {
       getConfiguration,
     } as unknown as IVSCodeWorkspace;
 
-    const agent = getHttpsProxyAgent(workspace);
+    const configOptions = getAxiosConfig(workspace, configuration, logger);
 
-    assert.deepStrictEqual(agent, undefined);
+    // should still set rejectUnauthorized flag
+    // @ts-ignore: cannot test options otherwise
+    assert.deepStrictEqual(configOptions.httpAgent?.options.rejectUnauthorized, proxyStrictSSL);
+    assert.deepStrictEqual(configOptions.httpsAgent?.options.rejectUnauthorized, proxyStrictSSL);
+  });
+
+  suite('.getAxiosConfig()', () => {
+    suite('when proxyStrictSsl is set', () => {
```

Comment:
How about rename from `when proxyStrictSsl is set` -> `when proxyStrictSsl is "true` to be clearer, as it's always set?

Reply:
No, from the user perspective it is not always set - technically, it may always be having a value, but that's not what I understand under set. I'll make it clearer, that the checkbox is meant by using "checked"